### ISSUE TITLE
Does not work for password containing the $ character

### ISSUE THREAD

**Issue by LouiseEH** [palantir/encrypted-config-value#120]
When encrypting password with the '$' character - the decrypted value of the password does not contain the part of the cleartext password from the '$' character and the remaining string.
If the password is e.g. "hello$world", the decrypted password is "hello".

**Comment by bavardage** [comment]
Would it be possible to provide more information about how you're interacting with ECV/the exact steps taken?

I'm unable to reproduce with a unit test - e.g the following passes:
```java
    @Test
    public void passwordContainingDollarRoundtrips() {
        String password = "hello$world";
        EncryptedValue encryptedValue = Algorithm.RSA.newEncrypter().encrypt(rsaPubKey, password);
        assertThat(encryptedValue.decrypt(rsaPrivKey), is(password));
    }
```

**Comment by LouiseEH** [comment]
Hi Bavardage,

I did the following:

1. Added jcenter() and:
compile group: 'com.palantir.config.crypto', name: 'encrypted-config-value-bundle-dropwizard1', version: '2.1.0'
to my build.gradle
2. Added:
bootstrap.addBundle(new EncryptedConfigValueBundle());
System.setProperty(KeyFileUtils.KEY_PATH_PROPERTY, "<the path to my key file>");
in my Application class
3. Executed:
java -jar <jar-file-name> generate-random-key -a AES -f encrypted-config-value.key
4. Executed:
java -jar <jar-file-name> encrypt-config-value -v <some-secret> -k encrypted-config-value.key
5. Added the output from 4 to my yml file, e.g. {$<output from step 4>}

When I tried starting up my application, it fails on the DB login where I used the encrypted password. I found a debug point where I could see the decrypted password, and it only contained the part of the password up to the $.
I'll try and see if I can get more information.
